# SARS-CoV-2 Bioinformatics & Data Science

A practical introduction to genomic surveillance using SARS-CoV-2 as an example and in the context of the FU course "SARS-CoV-2 Bioinformatics & Data Science" 2025.

## Schedule links for the workshop

* [2025-09-29 - Monday: Welcome, Linux re-cap, SARS-CoV-2 sequencing](#0)  
* [2025-09-30 - Tuesday: SARS-CoV-2 genome reconstruction](#6)  
* [2025-10-01 - Wednesday: SARS-CoV-2 nomenclature, mutation profiling & evolution](#7)  
* [2025-10-02 - Thursday: SARS-CoV-2 genome-based incidence estimation, phylogeny & outbreak investigation](#8)  

## Instructors

* Max von Kleist (FU, RKI), Martin Hölzer (RKI), Sofia Paraskevopoulou (RKI), Maria Trofimova (FU), Alexia Raharinirina (FU) & many great colleagues

## Schedule

> All events are held at FU Arnimallee 6, Room 017

### <a name="0"></a> Monday, 2025-09-29
| Time        | Welcome, Linux re-cap, SARS-CoV-2 sequencing |
| --          | --               |
| 10:00-10:30 | [Welcome & course intro](day-sc2-intro/README.md) |
| 10:30-11:00 | [RKI Genomic Surveillance and SARS-CoV-2](day-sc2-intro/README.md) |
| 11:00-12:00 | [Linux re-cap](day-welcome-linux-container-wms/linux.md) |
| 12:00-13:00 | Lunch break |
| 10:15-11:00 | [Sequencing (SARS-CoV-2)](day-sc2-seq-and-assembly/README.md) |
| 14:00-14:30 | Coffee break |
| 14:30-16:00 | [Setup & Hands-on](day-welcome-linux-container-wms/hands-on.md) |
| 16:00-16:15 | Wrap-up & questions |

### <a name="6"></a> Tuesday, 2025-09-30
| Time        | SARS-CoV-2 genome reconstruction |
| --          | --               |
| 10:00-10:15 | Debriefing previous day |
| 10:15-12:00 | [SARS-CoV-2 genome reconstruction](day-sc2-seq-and-assembly/README.md) |
| 12:00-13:00 | Lunch break |
| 13:00-14:30 | [Hands-on & demo](day-sc2-seq-and-assembly/hands-on.md) |
| 14:30-15:00 | Coffee break |
| 15:00-15:45 | Continue practical session |
| 15:45-16:00 | Wrap-up & questions |

### <a name="7"></a> Wednesday, 2025-10-01
| Time        | SARS-CoV-2 nomenclature, mutation profiling & evolution |
| --          | --               |
| 10:00-10:15 | Debriefing previous day |
| 10:15-11:30 | [SARS-CoV-2 nomenclature & mutation profiling](day-sc2-evolution/README.md) |
| 11:30-12:00 | [Hands-on & demo](day-sc2-evolution/hands-on.md) |
| 12:00-13:00 | Lunch break |
| 13:00-14:00 | [SARS-CoV-2 Evolution on a Dynamic Immune Landscape](day-sc2-incidence/README.md) |
| 14:30-15:00 | Coffee break |
| 15:00-15:45 | [Hands-on (SARS-CoV-2 Evolution on a Dynamic Immune Landscape)](day-sc2-incidence/hands-on_vasil.md) |
| 15:45-16:00 | Wrap-up & questions |

### <a name="8"></a> Thursday, 2025-10-02
| Time        | SARS-CoV-2 genome-based incidence estimation, phylogeny & outbreak investigation |
| --          | --               |
| 10:00-10:15 | Debriefing previous day |
| 10:15-11:15 | [SARS-CoV-2 pathogen evolution and genome-based incidence estimation](day-sc2-incidence/README.md) |
| 11:15-12:00 | [Hands-on & demo](day-sc2-incidence/hands-on_1_setup.md) |
| 12:00-13:00 | Lunch break |
| 13:00-13:30 | [Phylogeny and outbreak investigation](day-sc2-phylo-clustering/README.md) |
| 13:30-14:00 | [Outbreak detection & clustering via breakfast](day-sc2-phylo-clustering/README.md) |
| 14:00-14:30 | Coffee break |
| 14:30-15:45 | [Hands-on & demo](day-sc2-phylo-clustering/hands-on.md) |
| 15:45-16:00 | Wrap-up & questions |


### <a name="9"></a> Project work 2025-10-06 to 2025-10-09

### <a name="9"></a> Friday, 2025-10-10
| Time        | Project presentations |
| --          | --               |
| 10:00-11:00 | Project I |
| 11:00-12:00 | Project II |
| 12:00-13:00 | Lunch break |
| 13:00-14:00 | Project III |
| 14:00-15:00 | Project IV |
| 15:00-16:00 | Project V |
| 15:00-15:15 | Wrap-up |

## Acknowledgement

This course material is partly based on the following resources and on contributions from great people (no specific order):

* Martin Hölzer, RKI MF1, content about Linux, container, Nextflow, sequencing, genomic surveillance & glueing everything together
* Sebastian "Raverjay" Krautwurst, FSU Jena, some Linux and ONT content
* Stephan Fuchs, RKI MF1, some Linux and Assembly content 
* Matt Huska, RKI MF1, automatic test script for all md code blocks using [codedown](https://github.com/earldouglas/codedown) and general help
* Workshop structure inspired by [https://github.com/cinemaparis/2023](https://github.com/cinemaparis/2023)
* Max von Kleist, RKI P5 and FU Berlin, basically most content about sequencing and SC2 data science (evolution, epistasis, incidence esitmation, ...)
* Maureen Smith, Maria Trofimova, RKI P5 and FU Berlin, content on Incidence estimation
* Hugues Richard, RKI MF1, content about SC2 risk assessment
* Matt Huska & Denis Beslic, RKI MF1, content about SC2 outbreak detection & clustering
* We thank the [GISAID](https://gisaid.org) initiative for providing a framework for sharing SARS-CoV-2 sequence data internationally and all the submitting labs
